src/viem-config.ts

```diff
@@ -0,0 +1,41 @@
+import { Chain as ViemChain, defineChain } from 'viem';
+import * as viemChains from 'viem/chains';
+import { CHAINS } from './generated/chains';
+
+export function chains(): Array<ViemChain> {
+  const exportedChains: Array<ViemChain> = [];
+
+  for (const name in viemChains) {
+    const viemChain = (viemChains as { [name: string]: ViemChain })[name];
```

Comment:
@bbenligiray We'll need to decide what we want to do here as Viem doesn't support all of the same chains we have in our own `chains/` folder. We currently have 49 chains, with only 38 overlapping chains with Viem. i.e. this function will currently only output an array of length 38.

Currently, it re-uses whatever Viem has set for each chain (matching on chain ID) and then overwrites some URL fields to use what we've set.

I assume we'll want to add the remaining chains, but that will require adding more fields to whatever we currently have ([specifically adding fields like "nativeCurrency"](https://viem.sh/docs/clients/chains.html#custom-chains)). I wanted to get feedback first before going down that route, since it will be more work now (finding & setting the native currency/whatever else for 49 chains) and later (maintenance if these values ever change)

Reply:
We could also not merge with the `viem` chains at all and just use `defineChain` for everything, but we could potentially lose some information. e.g [`contracts`, `formatters`, `serializers`](https://github.com/wagmi-dev/viem/blob/42d74b93c10c71ef8a8310f866788407f7472e0c/src/chains/definitions/celo.ts) etc. I'm not sure how big of a deal that is

Reply:
I've been thinking about this a bit more and suspect we may we to use our own data, rather than merging with data from `viem/chains`

Reply:
@andreogle I have been using Viem in the Centurion app. It works _very_ well. Are any of our production  `@api3/chains` missing in **[Viem production chains list](https://github.com/wagmi-dev/viem/blob/main/src/chains/index.ts)** that you noticed?

Reply:
@andreogle I noticed this when I was in their docs. 

- https://viem.sh/docs/clients/chains.html#usage
- https://github.com/wagmi-dev/viem/blob/main/.github/CONTRIBUTING.md#chains

Reply:
I did a quick check and these are the chains currently listed in our `chains/` directory, but not included in `viem/chains`

```
Excluded chain id:43288 name:Boba/Avalanche
Excluded chain id:56288 name:Boba/BNB
Excluded chain id:71401 name:Godwoken testnet
Excluded chain id:71402 name:Godwoken
Excluded chain id:2221 name:Kava testnet
Excluded chain id:200101 name:Milkomeda C1 testnet
Excluded chain id:2001 name:Milkomeda C1
Excluded chain id:31 name:RSK testnet
Excluded chain id:30 name:RSK
Excluded chain id:647 name:SX Network testnet
Excluded chain id:416 name:SX Network
```

I'm not sure how successful we'll be trying to get 11 new chains included in `viem/chains` 🤔 

> I have been using Viem in the Centurion app. It works _very_ well.

Glad to hear it 😎

Reply:
> I've been thinking about this a bit more and suspect we may we to use our own data, rather than merging with data from viem/chains

I don't see any issues with this